platform/mime/core/platform-mime-core-impl/src/main/java/ddf/mime/mapper/MimeTypeMapperImpl.java

```diff
@@ -175,6 +179,26 @@ public String guessMimeType(InputStream is, String fileExtension)
         // from being used when a CustomMimeTypeResolver may be more appropriate.
         List<MimeTypeResolver> sortedResolvers = sortResolvers(mimeTypeResolvers);
 
+        if (StringUtils.isEmpty(fileExtension)) {
+            try (TemporaryFileBackedOutputStream tfbos = new TemporaryFileBackedOutputStream()) {
```

Comment:
Really? The interface looks like it does. Maybe the implementation doesn't?
https://docs.oracle.com/javase/8/docs/api/java/io/InputStream.html